include/libtorrent/torrent_handle.hpp

```diff
@@ -542,12 +542,20 @@ namespace aux {
 
 		// Returns true if this handle refers to a valid torrent and false if it
 		// hasn't been initialized or if the torrent it refers to has been
-		// aborted. Note that a handle may become invalid after it has been added
-		// to the session. Usually this is because the storage for the torrent is
-		// somehow invalid or if the filenames are not allowed (and hence cannot
-		// be opened/created) on your filesystem. If such an error occurs, a
-		// file_error_alert is generated and all handles that refers to that
-		// torrent will become invalid.
+		// removed from the session and destructed.
+                //
+                // A handle does not immediately become invalid after
+                // ``session::remove_torrent()`` completes; it will remain
+                // valid until after the last task (such as an I/O task)
+                // referring to the torrent has completed.
+                //
+                // To synchronize on whether the handle is in the session, use
+                // the expression ``session.find_torrent(handle.info_hash()) ==
+                // handle``. This expression will return true before
+                // ``torrent_removed_alert`` is posted, and false afterward.
+                //
+                // Clients should only use ``is_valid()`` to determine if the
+                // result of ``session::find_torrent()`` was successful.
```

Comment:
the indentation isn't right here. Also, maybe it would be better to just make all handles in alerts posted after torrent removal be invalid, and document that. I think it would be a simpler behavior at least, and probably more intuitive. What do you think?